src/main/java/ip/syssrc/array/SumOneThird.java

```diff
@@ -0,0 +1,49 @@
+package ip.syssrc.array;
+
+/**
+ * SumOneThird
+ *
+ * Assignment 3.3
+ *
+ * @author H071211074 - Wd. Ananda Lesmono <nandalesmono@gmail.com>
+ *
+ */
+public class SumOneThird {
+
+    public static void main(String[] args) {
+        int[] arr = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 100 };
+
+        
+        for (int index = 0; index < (arr.length / 3); index++) {
+            int sum = 0;
+            for (int j = 0; j < 3; j++ ) {
+                int aritmatika = arr.length / 3;
+                int deret = index + (j * aritmatika);
+                sum = sum + arr[deret];
+                if (j == 2) {
+                    System.out.println(arr[deret] + " = " + sum);;
+                } else {
+                    System.out.print(arr[deret] + " + ");
+                }
+            }
+        }
+        if (arr.length % 3 > 0) {
+            int sum = 0;
+            for (int index = ((arr.length / 3) * 3); index < arr.length; index++) {
+                sum = sum + arr[index];
+                if (index == arr.length - 1) {
+                    System.out.print(arr[index] + " = " + sum);
+                } else {
+                    System.out.print(arr[index] + " + ");
+                }
+            }
+        } 
+        System.out.print("\nsum of ");
+        int sum = 0;
+        for (int index = 0; index < arr.length; index++) {
+            System.out.print("[" + arr[index] + "]");
+            sum += arr[index];
+        }
+        System.out.println(" = " + sum);
```

Comment:
*main method* dari program ini hanya berisi kode berikut

```java
int[] arr = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15 };

for (int index = 0; index < (arr.length / 3); index++) {
// Your magic is here
}
```

Dan tidak ada kode tambahan selain:

- Mengganti komentar `// Your magic is here` dengan *logic* anda.
- Kode untuk mengatasi jika panjang array bukan kelipatan 3
- Kode untuk mencetak isi array beserta hasil penjumlahannya

Tugas ini tidak terdapat perulangan bersarang sama sekali, karena hal tersebut
membuat perhitungan jumlah dengan 1/3 panjang menjadi sia-sia *(Dimana tujuan
menggunakan 1/3 adalah untk menguranga iterasi yang harus dilakukan)*.

```java
        for (int index = 0; index < (arr.length / 3); index++) {
            int sum = 0;
            for (int j = 0; j < 3; j++ ) {
```

> Kode diatas meneyebabkan program melakukan iterasi sebanyak 3x 1/3 panjang
> array, dimana hal tersebut tidak sama dengan 1/3 panjang array

Proses mencetak penjumlahan setiap 3 elemennya terjadi didalam perulangan yang
sama dengan proses menghitung jumlahnya

```java
        if (arr.length % 3 > 0) {
            int sum = 0;
            for (int index = ((arr.length / 3) * 3); index < arr.length; index++) {
                sum = sum + arr[index];
                if (index == arr.length - 1) {
                    System.out.print(arr[index] + " = " + sum);
                } else {
                    System.out.print(arr[index] + " + ");
                }
            }
        }
```

> Kode diatas seharusnya tidak ada, karena proses mencetak penjumlahan setiap
> 3 elemennya juga terjadi di dalam perulangan sebelumnya (perulangan menghitung
> jumlah tiap 3 elemen, dimana `index < (arr.length / 3)`)